src/v0/destinations/af/transform.js

```diff
@@ -130,28 +130,29 @@ function getEventValueForUnIdentifiedTrackEvent(message) {
 
 function getEventValueMapFromMappingJson(message, mappingJson, isMultiSupport, config) {
   let eventValue = {};
-  const { addPropertiesAtRoot, afCurrencyAtRoot } = config;
-
+  const { addPropertiesAtRoot, afCurrencyAtRoot, listOfProps } = config;
+  const clonedProp = message.properties && lodash.cloneDeep(message.properties);
   if (addPropertiesAtRoot) {
-    eventValue = message.properties;
+    eventValue = clonedProp;
   } else {
-    set(eventValue, 'properties', message.properties);
+    if (Array.isArray(listOfProps) && listOfProps.length > 0) {
+      listOfProps.forEach((prop) => {
+        eventValue[prop.property] = clonedProp[prop.property];
+        delete clonedProp[prop.property];
+      });
+    }
+    set(eventValue, 'properties', clonedProp);
   }
 
   const sourceKeys = Object.keys(mappingJson);
   sourceKeys.forEach((sourceKey) => {
     set(eventValue, mappingJson[sourceKey], get(message, sourceKey));
```

Comment:
eventValue[mappingJson[sourceKey]] =  message[sourceKey];

mappingJson[sourceKey] does it contain complex values like a.b.c?

Reply:
I am not sure of this.